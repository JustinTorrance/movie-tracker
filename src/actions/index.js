export const loadMovies = (movies) => ({
  type: 'LOAD_MOVIES',
  movies
})

export const signIn = (user) => ({
  type: 'SIGN_IN',
  user
})

export const signOut = () => ({
  type: 'SIGN_OUT'
})

export const loadFavorites = (favorites) => ({
  type: 'LOAD_FAVORITES',
  favorites
})

export const isLoading = (bool) => ({
  type: 'IS_LOADING',
  bool
})