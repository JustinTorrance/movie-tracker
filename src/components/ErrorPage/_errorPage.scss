.error-page-container {
  width: 400px;
  margin: 200px auto;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;

  h1 {
    font-family: "Raleway", sans-serif;
    font-weight: 300;
    color: white;
    font-size: 4em;
    margin: 10px;
  }

  h2 {
    font-family: "Raleway", sans-serif;
    color: white;
    margin: 10px auto 30px;
  }

  button {
    background-color: #2D333B;
    border: 2px solid $color-blue;
    border-radius: 18px;
    color: #B7B7B7;
    font-family: $regular-font;
    font-size: 1.1em;
    padding: 6px 8px;
    transition: all .3s ease;
  
    &:hover {
      cursor: pointer;
      color: white;
      background-color: rgb(22, 131, 194);
      border-color: #2D333B
    }
  }
}