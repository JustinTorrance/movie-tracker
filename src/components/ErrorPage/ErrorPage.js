import React from 'react'
import { catchError } from '../../actions/index'
import { connect } from 'react-redux'

export const ErrorPage = ({returnToSite}) => {

  return (
    <div className='error-page-container'>
      <h1>OOPS!</h1>
      <h2>An unexpected error occurred...</h2>
      <button className='return-to-site-btn' onClick={() => returnToSite()}>Return to Site</button>
    </div>
  )
}

export const mapDispatchToProps = (dispatch) => ({
  returnToSite: () => dispatch(catchError(false))
})

export default connect(null, mapDispatchToProps)(ErrorPage)