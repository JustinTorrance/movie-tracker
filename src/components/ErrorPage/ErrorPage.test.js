import React from 'react'
import { shallow } from 'enzyme'
import { ErrorPage, mapDispatchToProps } from './ErrorPage'
import { catchError } from '../../actions/index'


describe('Loading', () => {
  let wrapper;
  const mockReturnToSite = jest.fn()

  beforeAll(() => {
    wrapper = shallow(
      <ErrorPage returnToSite={mockReturnToSite}/>
    )
  })

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should call returnToSite when the button is clicked', () => {
    wrapper.find('.return-to-site-btn').simulate('click')
    expect(mockReturnToSite).toHaveBeenCalled()
  })
})

describe('MapDispatchToProps', () => {
  it('Calls dispatch with a signOut action', () => {
    const mockDispatch = jest.fn()
    const actionToDispatch = catchError(false)
    const mappedProps = mapDispatchToProps(mockDispatch)
    mappedProps.returnToSite(false)
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })
})