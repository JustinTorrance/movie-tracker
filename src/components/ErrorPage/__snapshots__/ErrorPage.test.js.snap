// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should match the snapshot with all data passed in correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorPage
    returnToSite={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          OOPS!
        </h1>,
        <h2>
          An unexpected error occurred...
        </h2>,
        <button
          className="return-to-site-btn"
          onClick={[Function]}
        >
          Return to Site
        </button>,
      ],
      "className": "error-page-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "OOPS!",
        },
        "ref": null,
        "rendered": "OOPS!",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "An unexpected error occurred...",
        },
        "ref": null,
        "rendered": "An unexpected error occurred...",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Return to Site",
          "className": "return-to-site-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Return to Site",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            OOPS!
          </h1>,
          <h2>
            An unexpected error occurred...
          </h2>,
          <button
            className="return-to-site-btn"
            onClick={[Function]}
          >
            Return to Site
          </button>,
        ],
        "className": "error-page-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "OOPS!",
          },
          "ref": null,
          "rendered": "OOPS!",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "An unexpected error occurred...",
          },
          "ref": null,
          "rendered": "An unexpected error occurred...",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Return to Site",
            "className": "return-to-site-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Return to Site",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
